src/components/TooltipGlobal/TooltipGlobal.integration.test.js

```diff
@@ -0,0 +1,84 @@
+
+import { JetsTooltipGlobal } from './index';
+import { activeTooltipData, paramsData } from './__fixtures__';
+
+const setup = ({ data = {}, config = {}, params = {}, events = {} } = {}) => ({
+  user: userEvent.setup(),
+  ...render(
+    <MockJetsContextProvider config={config} params={{ ...paramsData(params) }} events={events}>
+      <JetsTooltipGlobal data={{ ...activeTooltipData(data) }} />
+    </MockJetsContextProvider>
+  ),
+});
+
+describe('JetsTooltipGlobal', () => {
+  describe('when a button in the tooltip is pressed', () => {
+    it('should fire onSeatSelect when "Select" is pressed', async () => {
+      const onSeatSelect = jest.fn();
+      const { user } = setup({ events: { onSeatSelect } });
+
+      // need to remove mocked functions from the expected object
+      const expectedData = { ...activeTooltipData() };
+      delete expectedData.seatmapElement;
+      delete expectedData.seatNode;
+
+      await user.click(screen.getByText(/Select/));
+      expect(onSeatSelect).toHaveBeenCalledTimes(1);
+      expect(onSeatSelect).toHaveBeenCalledWith(expect.objectContaining(expectedData));
+    });
+
+    it('should not fire onSeatSelect when seat selection is disabled and "Select" is pressed', async () => {
+      const onSeatSelect = jest.fn();
+      const { user } = setup({ events: { isSeatSelectDisabled: () => true, onSeatSelect } });
+
+      await user.click(screen.getByText(/Select/));
+      expect(onSeatSelect).not.toHaveBeenCalled();
+    });
+
+    it('should fire onSeatUnselect when "Unselect" is pressed', async () => {
+      const onSeatUnselect = jest.fn();
+      const { user } = setup({ data: { passenger: { passengerLabel: 'Dave Smith' } }, events: { onSeatUnselect } });
+
+      // need to remove mocked functions from the expected object
+      const expectedData = { ...activeTooltipData() };
+      delete expectedData.seatmapElement;
+      delete expectedData.seatNode;
+
+      await user.click(screen.getByText(/Unselect/));
+      expect(onSeatUnselect).toHaveBeenCalledTimes(1);
+      expect(onSeatUnselect).toHaveBeenCalledWith(expect.objectContaining(expectedData));
+    });
+
+    it('should not fire onSeatUnselect when the passenger is readOnly and "Unselect" is pressed', async () => {
+      const onSeatUnselect = jest.fn();
+      const { user } = setup({
+        data: { passenger: { passengerLabel: 'Dave Smith', readOnly: true } },
+        events: { isSeatSelectDisabled: () => true, onSeatUnselect },
+      });
+
+      await user.click(screen.getByText(/Unselect/));
+      expect(onSeatUnselect).not.toHaveBeenCalled();
+    });
+
+    it('should fire onTooltipClose when "Cancel" is pressed', async () => {
+      const onTooltipClose = jest.fn();
+      const { user } = setup({ events: { onTooltipClose } });
+
+      await user.click(screen.getByText(/Cancel/));
+      expect(onTooltipClose).toHaveBeenCalledTimes(1);
+      expect(onTooltipClose).toHaveBeenCalledWith(
```

Comment:
Not sure if this assertion is worth it as in this case the tooltip will be closed independently of the data being passed right? which seems a bit different from the select and unselect seats checks added in previous tests

Reply:
I did think about leaving it out but there is a callback prop (`onSeatMouseLeave`) associated with that data being passed so wanted to make sure that is working properly.